stats/record.go

```diff
@@ -60,10 +66,19 @@ func Record(ctx context.Context, ms ...Measurement) {
 // RecordWithTags is useful if you want to record with tag mutations but don't want
 // to propagate the mutations in the context.
 func RecordWithTags(ctx context.Context, mutators []tag.Mutator, ms ...Measurement) error {
+	return RecordWithTagsAndAttachments(ctx, mutators, nil, ms...)
+}
+
+// RecordWithTagsAndAttachments records measurements and the given exemplar attachments at once.
+//
+// Measurements will be tagged with the tags in the context mutated by the mutators.
+// RecordWithTags is useful if you want to record with tag mutations but don't want
+// to propagate the mutations in the context.
+func RecordWithTagsAndAttachments(ctx context.Context, mutators []tag.Mutator, attachments metricdata.Attachments, ms ...Measurement) error {
```

Comment:
For `Attachments` that makes sense, though I suggest we still keep the `recordWithTags` API since it covers most of the use cases and it's easier to use than creating `options`.

Reply:
SG

Reply:
Updated in https://github.com/census-instrumentation/opencensus-go/pull/1123/commits/0d5b2d89a3abe77a8a0add018991abc9290e3978.